### ISSUE TITLE
BUG

### ISSUE THREAD

**Issue by lucasjkr** [copona/copona#119]
Administrator should not be able to remove the ability to manage user groups from the Administrator user group... Oops.

**Comment by arnisjuraga** [comment]
You mean - Top admin user should not be allowed to remove himself from product groups?
Should we just create it by default - TOP admins role is not checked against user/group permissions, but has access to everything always?

**Comment by lucasjkr** [comment]
Yes, perhaps members of the Administrators user group have access to all permissions would be the best solution. If people want more granular permissions, they need to make an additional group besides Administrators and Demonstration which has the privs they want to assign.

**Comment by arnisjuraga** [comment]
Simple fix, if you are ok with this - "top admin" is checked against `user_id==1` 
https://github.com/copona/copona/commit/9d54a914a88bff9a2d9c07fd3f30ebe6d22028ed#diff-660bb014fec8cc2feed776537eada665R92

**Comment by ADDCreative** [comment]
Would not a better way to do this be to use the user_group_id. That way with a bit more modification you can make the permissions uneditable and always selected for the top admin group in user/user_permission/edit. It will then be clearer who has what permissions.

**Comment by arnisjuraga** [comment]
So, you suggest to disable permission edits for Administator (group_id == 1) group?

**Comment by lucasjkr** [comment]
Yeah, it shouldn't be based on the user ID, it should be anyone in the Administrators group. If someone needs fewer permissions, they should be in a different group.

**Comment by ADDCreative** [comment]
Yes, that way it's clear that top admis will always have full permissions. At the moment you could have two users in the admin group (say user_id = 1 and user_id = 2), but there permissions could be different. You also have the issue that user_id = 1 can be deleted, where it's a bit harder to delete the top admin group as it has to be empty.

**Comment by arnisjuraga** [comment]
If we stick that to the group, then again - it must be hard-coded to group id. But "super admin" user, let's say - user with "id==1" - still will be able to create new groups and assign himself to different groups and remove Administrators group. 

If we limit these global admin permissions to group, then
- we should disable ANY user of Administrator group to change this group permissions as they will not have any impact. It will be confusing that any Admin user can change Administrator group permissions, but that does not change real permissions.  

That's why I chose user_id = 1, 
this means, that:
- we hard code it to only one specific user
- a user can still be renamed
- user exists in the Database table, so the password can be easily recovered directly in DB
- "admin" user cannot mess with a permissions if he changes his group membership to some other group (but of course, this is also some security issue, because if he changes the group to user_id==1, and then creates new user_id==2 assigning him to Administrator group, then in fact - user_id==1 will still have Full permissions, no matter in which group he is)
- and we have only one "hard coded" top-admin limitations.

On the other hand, if change this approach to Administrator group, then:
- Administrator group members must be not removable.
- Administrator group permissions must be not changeable.

Opinions?

**Comment by ADDCreative** [comment]
So if making a super admin you would need to.
- Make super admin have all permissions (as you have done)
- Prevent super admin from being deleted
- Have some way of showing the super admin always has all permissions whatever group they are in.

If making a super admin group you would need to.
- Make super admin group have all permissions
- Disable the editing of permissions for super admin group
- Prevent super admin from having less than one user in it (this also prevents the super admin group from being deleted)

Both have advantage and disadvantages, so whatever you feel is easy to implement and clear for users.

**Comment by lucasjkr** [comment]
I Like @ADDCreative's solution myself - before deleting a user, first check if they're a member of the admin group, and if they are, go on to check that the admin group will still have a member who's account is enabled. (Basically, make sure the count of the admin group is at least 2 before doing the deletion)
